pkg/registry/apis/provisioning/resources/dualwriter.go

```diff
@@ -131,6 +133,24 @@ func (r *DualReadWriter) Delete(ctx context.Context, opts DualWriteOptions) (*Pa
 	return parsed, err
 }
 
+func (r *DualReadWriter) getConfiguredBranch() string {
```

Comment:
This normally done transparently for each implementation. Each implementation will check with the corresponding config.

Reply:
ok, I see. The expectation is that frontend only specifies the branch when it's not the configured one (we don't enforce it currently on the API). In that way you only need to compare with empty string as we do for removing resources a couple of lines below.